Source/Core/DolphinQt2/GameList/GameList.cpp

```diff
@@ -138,6 +152,53 @@ void GameList::SetDefaultISO()
   Settings().SetDefaultGame(GetSelectedGame());
 }
 
+void GameList::OpenContainingFolder()
+{
+  std::cout << GameFile(GetSelectedGame()).GetInternalName().toStdString() << std::endl;
+  QString url = QStringLiteral("file://").append(QFileInfo(GetSelectedGame()).dir().absolutePath());
+  QDesktopServices::openUrl(QUrl(url));
+}
+
+void GameList::DeleteFile()
+{
+  QMessageBox confirm_dialog(this);
+
+  confirm_dialog.setIcon(QMessageBox::Warning);
+  confirm_dialog.setText(tr("Are you sure you want to delete this file?"));
+  confirm_dialog.setInformativeText(tr("You won't be able to undo this!"));
+  confirm_dialog.setStandardButtons(QMessageBox::Yes | QMessageBox::Cancel);
+
+  if (confirm_dialog.exec() == QMessageBox::Yes)
+  {
+    bool deletion_successful = false;
+
+    while (!deletion_successful)
+    {
+      deletion_successful = File::Delete(GetSelectedGame().toStdString());
+
+      if (deletion_successful)
+      {
+        // TODO Reload the game list
```

Comment:
Rather than force-update the whole list manually for disk system changes, it would be preferable to use [QFileSystemWatcher](http://doc.qt.io/qt-5/qfilesystemwatcher.html) to monitor the game folder/whatever is being watched, and just handle the signals it events for file changes respectively. This way, you get proper list updating whenever a user deletes a file directly instead of using the application to do it as well.